[qcoro] New port

danvratil opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

danvratil committed:
[qcoro-qt6] New port

danvratil commented:
I disabled support for OSX for now, because Apple Clang compiler ICEs in Release build, I need to reproduce it in upstream CI and debug the problem further to be able to find out how to workaround the ICE.

danvratil committed:
Rename the port to qcoro

Adela0814 commented on ports/qcoro/portfile.cmake:
```diff
@@ -0,0 +1,57 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO danvratil/qcoro
+    REF "v0.9.0"
```
```suggestion
    REF "v${VERSION}"
```

danvratil replied:
Removed.

Adela0814 commented on ports/qcoro/0001-qt6-deprecated-qwebsocket-error.patch:
```diff
@@ -0,0 +1,32 @@
+commit 1f420bb8f9f0028d1289143c4a51bcf8da88387d
+Author: Marius P <nmariusp1@gmail.com>
+Date:   Sat May 13 12:40:48 2023 +0300
+
+    qcorowebsocket replace QWebSocket::error with QWebSocket::errorOccurred
+    
+    I build using cmake build configuration "Debug" not "RelWithDebInfo".
+    Building qcoro fails with error "qcoro/qcoro/websockets/qcorowebsocket.cpp:
+    35:73: error: ‘void QWebSocket::error(QAbstractSocket::SocketError)’
+    is deprecated: Use errorOccurred instead [-Werror=deprecated-declarations]".
+    Because "# Only enable strict warnings in debug mode
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wall -Wextra -Werror -pedantic")".
+
```
Please remove these lines.

Adela0814 commented on ports/qcoro/vcpkg.json:
```diff
@@ -0,0 +1,57 @@
+  "homepage": "https://www.github.com/danvratil/qcoro",
+  "documentation": "https://qcoro.dvratil.cz",
+  "license": "MIT",
+  "supports": "!osx",
```
Please move this to `ci. baseline` instead of in the `supports` field. Because upstream supports `OSX`.

danvratil replied:
Done, hopefully I did it correctly.

Adela0814 commented:
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

Neumann-A commented on ports/qcoro/0001-qt6-deprecated-qwebsocket-error.patch:
```diff
@@ -0,0 +1,32 @@
+index be9c1f7..c686d14 100644
+--- a/qcoro/websockets/qcorowebsocket.cpp
++++ b/qcoro/websockets/qcorowebsocket.cpp
+@@ -32,7 +32,13 @@ public:
```
Github user names checks out to be upstream?

danvratil replied:
I'm the upstream :) This patch is already in the main branch, just wasn't included in 0.9.0 release. It will be dropped once 0.10.0 is out.

danvratil committed:
Clean up upstream patch

danvratil committed:
Add x86-osx=failed to ci.baseline

Triggers ICE in release build, but OSX is officially supported,
so just expect CI to fail.

danvratil committed:
Use ${VERSION} in portfile

danvratil committed:
Update version

Adela0814 commented:
@danvratil I encountered an error while testing usage:
```
1> [CMake] CMake Error at E:/4/vcpkg/scripts/buildsystems/vcpkg.cmake:855 (_find_package):
1> [CMake]   By not providing "FindQCoro6Coro.cmake" in CMAKE_MODULE_PATH this project
1> [CMake]   has asked CMake to find a package configuration file provided by
1> [CMake]   "QCoro6Coro", but CMake did not find one.
1> [CMake] 
1> [CMake]   Could not find a package configuration file provided by "QCoro6Coro" with
1> [CMake]   any of the following names:
1> [CMake] 
1> [CMake]     QCoro6CoroConfig.cmake
1> [CMake]     qcoro6coro-config.cmake
1> [CMake] 
1> [CMake]   Add the installation prefix of "QCoro6Coro" to CMAKE_PREFIX_PATH or set
1> [CMake]   "QCoro6Coro_DIR" to a directory containing one of the above files.  If
1> [CMake]   "QCoro6Coro" provides a separate development package or SDK, be sure it has
1> [CMake]   been installed.
1> [CMake] Call Stack (most recent call first):
1> [CMake]   C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.24/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
1> [CMake]   E:/4/vcpkg/installed/x64-windows/share/QCoro6Core/QCoro6CoreConfig.cmake:30 (find_dependency)
1> [CMake]   E:/4/vcpkg/scripts/buildsystems/vcpkg.cmake:855 (_find_package)
1> [CMake]   CMakeTest/CMakeLists.txt:15 (find_package)
1> [CMake] -- Configuring incomplete, errors occurred!
1> [CMake] See also "E:/CMakeTest/out/build/x64-debug/CMakeFiles/CMakeOutput.log".
1> [CMake] See also "E:/CMakeTest/out/build/x64-debug/CMakeFiles/CMakeError.log".
1> 'C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="E:\CMakeTest\out\install\x64-Debug" -DCMAKE_C_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe" -DCMAKE_CXX_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe"  -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" -DCMAKE_TOOLCHAIN_FILE="E:/4/vcpkg/scripts/buildsystems/vcpkg.cmake" "E:\CMakeTest" 2>&1"' execution failed with error: ''C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="E:\CMakeTest\out\install\x64-Debug" -DCMAKE_C_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe" -DCMAKE_CXX_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe"  -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" -DCMAKE_TOOLCHAIN_FILE="E:/4/vcpkg/scripts/buildsystems/vcpkg.cmake" "E:\CMakeTest" 2>&1"' returned with exit code: 1'.
```

Usage:
```
qcoro-qt6 provides CMake targets:

    find_package(QCoro6Core CONFIG REQUIRED)
    target_link_libraries(main PRIVATE QCoro6::Core)

    find_package(QCoro6DBus CONFIG REQUIRED)
    target_link_libraries(main PRIVATE QCoro6::DBus)

    find_package(QCoro6Network CONFIG REQUIRED)
    target_link_libraries(main PRIVATE QCoro6::Network)

    find_package(QCoro6Qml CONFIG REQUIRED)
    target_link_libraries(main PRIVATE QCoro6::Qml)

    find_package(QCoro6Quick CONFIG REQUIRED)
    target_link_libraries(main PRIVATE QCoro6::Quick)

    find_package(QCoro6Test CONFIG REQUIRED)
    target_link_libraries(main PRIVATE QCoro6::Test)

    find_package(QCoro6WebSockets CONFIG REQUIRED)
    target_link_libraries(main PRIVATE QCoro6::WebSockets)
```

Adela0814 commented:
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

danvratil committed:
Add missing QCoro6Coro cmake module

danvratil committed:
Update version

Adela0814 commented:
All feature tested pass on the following triplets:

- x64-windows
- x64-windows-static
- x86-windows

Usage tested pass on `x64-windows`.

BillyONeal commented:
Thanks for the new port and 'confirming with upstream' that the patch was OK :)
